snappy_pipeline/workflows/adapter_trimming/__init__.py

```diff
@@ -0,0 +1,676 @@
+      complexity_threshold: 30            # the threshold for low complexity filter (0~100). Default is 30, which means 30% complexity is required. (int [=30])
+      filter_by_index1: ""                # specify a file contains a list of barcodes of index1 to be filtered out, one barcode per line (string [=])
+      filter_by_index2: ""                # specify a file contains a list of barcodes of index2 to be filtered out, one barcode per line (string [=])
+      filter_by_index_threshold: 0        # the allowed difference of index barcode for index filtering, default 0 means completely identical. (int [=0])
+      correction: False                   # enable base correction in overlapped regions (only for PE data), default is disabled
+      overlap_len_require: 30             # the minimum length to detect overlapped region of PE reads. This will affect overlap analysis based PE merge, adapter trimming and correction. 30 by default. (int [=30])
+      overlap_diff_limit: 5               # the maximum number of mismatched bases to detect overlapped region of PE reads. This will affect overlap analysis based PE merge, adapter trimming and correction. 5 by default. (int [=5])
+      overlap_diff_percent_limit: 20      # the maximum percentage of mismatched bases to detect overlapped region of PE reads. This will affect overlap analysis based PE merge, adapter trimming and correction. Default 20 means 20%. (int [=20])
+      umi: False                          # enable unique molecular identifier (UMI) preprocessing
+      umi_loc: ""                         # specify the location of UMI, can be (index1/index2/read1/read2/per_index/per_read, default is none (string [=])
+      umi_len: 0                          # if the UMI is in read1/read2, its length should be provided (int [=0])
+      umi_prefix: ""                      # if specified, an underline will be used to connect prefix and UMI (i.e. prefix=UMI, UMI=AATTCG, final=UMI_AATTCG). No prefix by default (string [=])
+      umi_skip: 0                         # if the UMI is in read1/read2, fastp can skip several bases following UMI, default is 0 (int [=0])
+      overrepresentation_analysis: False  # enable overrepresented sequence analysis.
+
+""".lstrip()
+
+
+class AdapterTrimmingStepPart(BaseStepPart):
+    """Adapter trimming common features"""
+
+    #: Step name
+    name = ""
+
+    #: Class available actions
+    actions = ("run",)
+
+    def __init__(self, parent):
+        super().__init__(parent)
+        self.base_path_in = "work/input_links/{library_name}"
+        self.base_path_out = "work/{trimmer}.{{library_name}}"
+        #: Path generator for linking in
+        self.path_gen = LinkInPathGenerator(
+            self.parent.work_dir,
+            self.parent.data_set_infos,
+            self.parent.config_lookup_paths,
+            preprocessed_path=self.config["path_link_in"],
+        )
+
+    @classmethod
```

Comment:
Inheriting from instance method: `BaseStepPart.get_input_files()`. Plus no need to change Class state.

Reply:
I don't understand: the [parent method](https://github.com/bihealth/snappy-pipeline/blob/fb39e0a356a571d0101fa84fcb8040c7ec2be974/snappy_pipeline/workflows/abstract/__init__.py#L127) must be overridden, as far as I understand